### PR TITLE
Module-specific docs links

### LINKED ISSUE #128: Creating javadoc for each module
An aggregate Javadoc is created and linked in the website.  However, we do not generate and link specific Javadocs for the modules (i.e. gcloud-java-core, gcloud-java-datastore, etc).

### LINKED ISSUE #228: Modify the title, description and javadoc reference for each module
Currently the `README.md` for each module contains the same title and description
as the main `README.md` page. 

We should change it to be more specific to the service.
e.g. for Datastore

**Current Title:** Google Cloud Java Client
**Proposed Title:**  Google Cloud Java Client for Datastore

**Current Description:** Java idiomatic client for Google Cloud Platform services.
**Proposed Description:** Java idiomatic client for Google Cloud Datastore.
* Also, change link reference.

Also, we can remove the "This client supports Google Cloud Datastore" statement for the module
and we should update the javadoc reference to the specific service, similar to what we do for example
(e.g. Datastore reference should be something like http://googlecloudplatform.github.io/gcloud-java/0.0.7/apidocs/index.html?com/google/gcloud/datastore/package-summary.html).

### PR DESCRIPTION
**Pull request by ajkannan**
Fixes #128
Fixes #228 

Adds links to specific modules from service-specific READMEs and service-specific maven sites.

### CODE REVIEW

**Comment by ajkannan** [comment]
@aozarov: it looks like there isn't a boilerplate way to add a link on the left of the maven-generated site since it isn't one of the report goals listed [here] (https://maven.apache.org/plugins/maven-project-info-reports-plugin/plugin-info.html).  Let me know if you know of another way that isn't too much effort.

File: gcloud-java-core/README.md [hunk 1]
```diff
@@ -1,17 +1,15 @@
-Google Cloud Java Client
-==========================
+Google Cloud Java Client -- Core Services
```
Comment (aozarov) [inline gcloud-java-core/README.md]: Just -- Core or -- Common is enough

File: gcloud-java-storage/README.md [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
-Java idiomatic client for [Google Cloud Platform][cloud-platform] services.
+Java idiomatic client for [Google Cloud Storage] (https://cloud.google.com/storage/). 
 
 [![Build Status](https://travis-ci.org/GoogleCloudPlatform/gcloud-java.svg?branch=master)](https://travis-ci.org/GoogleCloudPlatform/gcloud-java)
 [![Coverage Status](https://coveralls.io/repos/GoogleCloudPlatform/gcloud-java/badge.svg?branch=master)](https://coveralls.io/r/GoogleCloudPlatform/gcloud-java?branch=master)
 
 -  [Homepage] (https://googlecloudplatform.github.io/gcloud-java/)
--  [API Documentation] (http://googlecloudplatform.github.io/gcloud-java/apidocs)
+-  [API Documentation] (http://googlecloudplatform.github.io/gcloud-java/apidocs/index.html?com/google/gcloud/storage/package-summary.html)
 -  [Examples] (http://googlecloudplatform.github.io/gcloud-java/apidocs/index.html?com/google/gcloud/examples/package-summary.html)
```
Comment (aozarov) [inline gcloud-java-storage/README.md]: Maybe http://googlecloudplatform.github.io/gcloud-java/0.0.7/apidocs/?com/google/gcloud/examples/StorageExample.html ?

File: gcloud-java-core/README.md [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
-Java idiomatic client for [Google Cloud Platform][cloud-platform] services.
+This module provides common functionality required by service-specific modules of this library.
 
 [![Build Status](https://travis-ci.org/GoogleCloudPlatform/gcloud-java.svg?branch=master)](https://travis-ci.org/GoogleCloudPlatform/gcloud-java)
 [![Coverage Status](https://coveralls.io/repos/GoogleCloudPlatform/gcloud-java/badge.svg?branch=master)](https://coveralls.io/r/GoogleCloudPlatform/gcloud-java?branch=master)
 
 -  [Homepage] (https://googlecloudplatform.github.io/gcloud-java/)
--  [API Documentation] (http://googlecloudplatform.github.io/gcloud-java/apidocs)
+-  [API Documentation] (http://googlecloudplatform.github.io/gcloud-java/apidocs/index.html?com/google/gcloud/package-summary.html)
 -  [Examples] (http://googlecloudplatform.github.io/gcloud-java/apidocs/index.html?com/google/gcloud/examples/package-summary.html)
```
Comment (aozarov) [inline gcloud-java-core/README.md]: http://googlecloudplatform.github.io/gcloud-java/0.0.7/apidocs/?com/google/gcloud/examples/DatastoreExample.html

**Comment by aozarov** [comment]
@ajkannan I merged it before giving you a chance to react to my suggestions. apologies.
If you agree with the suggestions do you mind to send a new PR for it?

**Comment by ajkannan** [comment]
Yes, I agree with the suggestions.  I'll create a new PR

**Pull request merged**